stumpy/snippets.py

```diff
@@ -83,22 +94,25 @@ def _get_all_profiles(
             f"Please try `m <= len(T) // 2`."
         )
 
+    if s is not None:
+        s = min(int(s), m)
+    else:
+        percentage = np.clip(percentage, 0.0, 1.0)
+        s = min(math.ceil(percentage * m), m)
+
     right_pad = 0
+    T_subseq_isconstant = core.rolling_isconstant(T, s, T_subseq_isconstant)
```

Comment:
I think we need to compute `T_subseq_isconstant` before extending `T`. This is because the user might have provided the value of `T_subseq_isconstant`. If the provided value is a boolean array, then it has shape `len(T) - s + 1`. So, I think we need to compute it here, and then extends it in the if-block below by padding it with `False`.

Reply:
I don't know :) I trust your judgement here but please double check

Reply:
I will check it again.